src/main/java/com/thebois/models/beings/AbstractBeingGroup.java

```diff
@@ -2,13 +2,14 @@
 
 import java.util.ArrayList;
 import java.util.Collection;
+import java.util.concurrent.ConcurrentLinkedQueue;
 
 /**
  * An abstract implementation of IBeingGroup.
  */
 public abstract class AbstractBeingGroup implements IBeingGroup {
 
-    private Collection<IBeing> beings = new ArrayList<>();
+    private final ConcurrentLinkedQueue<IBeing> beings = new ConcurrentLinkedQueue<>();
```

Comment:
Question
What does this change do?

Reply:
`ConcurrentLinkedQueue` implements `Collection` so you shouldn't change the type of the field. Just swap out the `= new ArrayList<>();` part.

Reply:
Done 👍